version: "3"
networks:
    my_network:
      driver: bridge
services:
  nginx:
    image: nginx:latest
    ports:
      - "3000:3000"
      - "443:443"
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf
      - ./certs:/etc/nginx/certs
    depends_on:
      - app
    networks:
      - my_network
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: udv-bff-app
    restart: always
    image: udv-app:v1.0.0
    ports:
      - "3001:3000"
    environment:
      PORT: ${PORT}
      PG_HOST: udv-backend-postgres
      PG_DATABASE: ${PG_DATABASE}
      PG_USER: ${PG_USER}
      PG_PASSWORD: ${PG_PASSWORD}
      PG_PORT: ${PG_PORT}
      DATABASE_URL: ${DATABASE_URL}
    depends_on:
      - db
    networks:
      - my_network
  db:
    image: postgres:latest
    container_name: udv-postgres
    restart: always
    environment:
      POSTGRES_DB: ${PG_DATABASE}
      POSTGRES_USER: ${PG_USER}
      POSTGRES_PASSWORD: ${PG_PASSWORD}
    hostname: ${PG_HOST}
    ports:
      - "${PG_PORT}:${PG_PORT}"
    healthcheck:
      test:
        [
          "CMD-SHELL",
          "pg_isready -U ${PG_USER} -d ${PG_DATABASE} -h ${PG_HOST} -p ${PG_PORT}",
        ]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - my_network
      
  
